frontend/app/performance/lighthouse/run.mjs

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+// @ts-check
+import fs from "fs"
+import path from "path"
+
+import { LighthouseOrchestrator } from "./LighthouseOrchestrator.mjs"
+import {
+  MODES,
+  PERFORMANCE_APPS_DIRECTORY,
+  STREAMLIT_ROOT,
+} from "./constants.mjs"
+
+/**
+ * Run the Lighthouse performance test suite.
+ * @param {string} appsDirectory The directory containing the streamlit apps to
+ * run
+ */
+const run = async appsDirectory => {
+  const now = new Date()
+    .toISOString()
+    .replace(/[-:.TZ]/g, "")
+    .slice(0, 15)
+    .replace("T", "-")
```

Comment:
nit: could you add a short comment explaining what the output of the transformation here is?